src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/AllocHGlobalTests.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class AllocHGlobalTests
+    {
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(100)]
+        public void AllocHGlobal_Int32_ReadableWritable(int size)
```

Comment:
This test isn't needed. Many tests under `src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/` depend on this functionality and do this exact scenario.